src/Symfony/Component/Routing/Generator/UrlGeneratorInterface.php

```diff
@@ -71,16 +71,12 @@ interface UrlGeneratorInterface extends RequestContextAwareInterface
      *
      * The special parameter _fragment will be used as the document fragment suffixed to the final URL.
      *
-     * @param string $name          The name of the route
-     * @param mixed  $parameters    An array of parameters
```

Comment:
That one confused me a bit. `$parameters` is documented as `mixed`, suggesting we could pass something other than an array (`ArrayObject` maybe?).

However, our default implementation of the interface is only tested with arrays. If I pass something else, I run into errors because we call array functions on the value passed as `$parameters`.

This is why I decided to add the `array` type-hint here.

Reply:
I think the phpdoc might have meant to be `mixed[]`

Reply:
True. I'll work out a PR to fix this on older branches.

Reply:
see #32185